modal/image.py

```diff
@@ -650,6 +650,24 @@ def my_build_function():
         """
         return self.extend(build_function=(raw_function, kwargs))
 
+    def env(self, vars: dict[str, str]) -> "_Image":
+        """Sets the environmental variables of the image.
+
+        **Example**
+
+        ```python
+        image = (
+            modal.Image.conda()
+                .env({"CONDA_OVERRIDE_CUDA": "11.2"})
+                .conda_install("jax", "cuda-nvcc", channels=["conda-forge", "nvidia"])
+                .pip_install("dm-haiku", "optax")
+        )
+        ```
+        """
+        return self.extend(
+            dockerfile_commands=["FROM base"] + [f"ENV {key}={shlex.quote(val)}" for (key, val) in vars.items()]
```

Comment:
I think if `val` has newlines, this will literally put the string `\n` (two characters: backslash, n) inside the environment variable. But it also seems uncommon, so it's probably okay. We can always fix it later since this is just in the client library.